rewrite-java/src/main/java/org/openrewrite/java/ReplaceStringLiteralWithConstant.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+
+import java.util.List;
+
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ReplaceStringLiteralWithConstant extends Recipe {
+
+    @Option(displayName = "Fully Qualified Name of the constant",
+            example = "org.springframework.http.MediaType.APPLICATION_JSON_VALUE")
+    String fullyQualifiedConstantName;
+
+    @Option(displayName = "(Optional) String literal value that needs to be replaced. Only String literal accepted.",
+            example = "application/json",
+            required = false )
+    @Nullable
+    String literalValue;
+
+    public ReplaceStringLiteralWithConstant(String fullyQualifiedNameOfConst) {
+        this.fullyQualifiedNameOfConst = fullyQualifiedNameOfConst;
+        this.literalValue = fullyQualifiedNameOfConst.substring(fullyQualifiedNameOfConst.lastIndexOf(".")+1);
+    }
+
+    public ReplaceStringLiteralWithConstant(String fullyQualifiedNameOfConst, String literalValue) {
+        this.fullyQualifiedNameOfConst = fullyQualifiedNameOfConst;
+        this.literalValue = literalValue;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Replace String Literal with Constant";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace a String Literal with named constant. If String literal value no provided, the recipe will use constant's Name in place of literal value.";
+    }
+
+    @Override
+    public JavaVisitor<ExecutionContext> getVisitor() {
+        return new JavaVisitor<ExecutionContext>() {
+
+            @Nullable
+            J.FieldAccess constant;
+
+            @Override
+            public J visitLiteral(J.Literal literal, ExecutionContext executionContext) {
+                if(isStringLiteral(literal)) {
+                    String owningType = fullyQualifiedNameOfConst.substring(0,fullyQualifiedNameOfConst.lastIndexOf("."));
+                    maybeAddImport(owningType,false);
+                    return buildConstant().withPrefix(literal.getPrefix());
```

Comment:
Have you tried using `JavaTemplate` to construct the replacement for the string literal? It should be able to accomplish the same thing as `buildConstant()` in fewer lines of code.  See [this document](https://docs.openrewrite.org/authoring-recipes/modifying-methods-with-javatemplate) for some examples of how to use it. Give it a try and let me know if you have questions.